controllers/metal3labelsync_controller_test.go

```diff
@@ -329,6 +334,544 @@ var _ = Describe("Metal3LabelSync controller", func() {
+					Name: "testNode",
+				}}).CoreV1()
+				mlsReconcile := &Metal3LabelSyncReconciler{
+					Client:         c,
+					ManagerFactory: baremetal.NewManagerFactory(c),
+					Log:            klogr.New(),
+					CapiClientGetter: func(ctx context.Context, c client.Client, cluster *capi.Cluster) (
+						clientcorev1.CoreV1Interface, error,
+					) {
+						return corev1Client, nil
+					},
+					WatchFilterValue: "",
+				}
+				req := reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      "testHost",
+						Namespace: namespaceName,
+					},
+				}
+				result, err := mlsReconcile.Reconcile(context.TODO(), req)
+
+				if tc.expectError {
+					Expect(err).To(HaveOccurred())
+				} else {
+					Expect(err).NotTo(HaveOccurred())
+				}
+				if tc.expectRequeue {
+					Expect(result.Requeue || result.RequeueAfter > 0).To(BeTrue())
+				} else {
+					Expect(result.Requeue || result.RequeueAfter > 0).To(BeFalse())
+				}
+
+				node, _ := corev1Client.Nodes().Get(context.TODO(), "testNode", metav1.GetOptions{})
+				Expect(node.Labels).To(Equal(tc.expectLabelsync))
+
+			},
+			Entry("Baremetal host not found", testCaseReconcile{
+				expectError:   false,
+				expectRequeue: false,
+			}),
+			Entry("Paused Baremetal", testCaseReconcile{
+				host: &bmh.BareMetalHost{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testHost",
+						Namespace: namespaceName,
+						Annotations: map[string]string{
+							"baremetalhost.metal3.io/paused": "",
+						},
+					},
+				},
+				expectRequeue: true,
+			}),
+			Entry("Baremetal host with no ConsumerRef", testCaseReconcile{
+				host: &bmh.BareMetalHost{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testHost",
+						Namespace: namespaceName,
+					},
+				},
+			}),
+			Entry("Unknown GroupVersionKind in BareMetalHost ConsumerRef", testCaseReconcile{
+				host: &bmh.BareMetalHost{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testHost",
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							"foo.metal3.io/bar": "blue",
+						},
+					},
+					Spec: bmh.BareMetalHostSpec{
+						ConsumerRef: &corev1.ObjectReference{
+							Name:       metal3machineName,
+							Namespace:  namespaceName,
+							Kind:       "notMetal3Machine",
+							APIVersion: "not" + capm3.GroupVersion.String(),
+						},
+					},
+				},
+			}),
+			Entry("Could not find associated Metal3Machine", testCaseReconcile{
+				host: &bmh.BareMetalHost{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testHost",
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							"foo.metal3.io/bar": "blue",
+						},
+					},
+					Spec: bmh.BareMetalHostSpec{
+						ConsumerRef: &corev1.ObjectReference{
+							Name:       metal3machineName,
+							Namespace:  namespaceName,
+							Kind:       "Metal3Machine",
+							APIVersion: capm3.GroupVersion.String(),
+						},
+					},
+				},
```

Comment:
how about defining BMH the same way as `newMetalMachine` and reuse it in multiple places (at least 6-7) so we could get rid of the repetitive host structure?

Reply:
Thanks @furkatgofurov7 for the suggestion that has improved the code quality please check